src/alexapi/playback_handlers/vlchandler.py

```diff
@@ -191,16 +115,13 @@ def state_callback(self, event, player): # pylint: disable=unused-argument
 		if state in [vlc.State.Playing, vlc.State.Stopped, vlc.State.Ended, vlc.State.Error]:
 
 			report = {
-				vlc.State.Playing: ("STARTED", "PLAYING", self.stream_id),
-				vlc.State.Stopped: ("INTERRUPTED", "IDLE", self.stream_id),
-				vlc.State.Ended: ("FINISHED", "IDLE", self.stream_id),
-				vlc.State.Error: ("ERROR", "IDLE", self.stream_id)
+				vlc.State.Playing: self.report_play,
+				vlc.State.Stopped: self.report_stop,
+				vlc.State.Ended: self.report_finish,
+				vlc.State.Error: self.report_error
 			}
 
-			rThread = threading.Thread(target=self.__callback_report, args=report[state])
-			rThread.start()
+			report[state]()
```

Comment:
If you look at commit 4e42544ba121c1a367d7343a4386b9d6b42144af you will see that the reporting to AVS is still async. Only the platform calls are blocking now which is likely the correct behavior. The comment on that commit says a little more